plugins/package-managers/bazel/src/main/kotlin/CompositeBazelModuleRegistryService.kt

```diff
@@ -0,0 +1,102 @@
+ * [packagesPerRegistry] is a map of [BazelModuleRegistryService] instances to the list of packages they are
+ * responsible for.
+ */
+class CompositeBazelModuleRegistryService(
+    private val packagesPerRegistry: Map<BazelModuleRegistryService, List<String>>
+) : BazelModuleRegistryService {
+    companion object {
+        // A regular expression to extract the server and package parts of a registry URL.
+        internal val URL_REGEX = "^(?<server>.*/)modules/(?<package>[^/]+)/[^/]+/source\\.json$".toRegex()
+
+        /**
+         * Create a Composite Bazel Module Registry client instance.
+         */
+        fun create(urls: Set<String>, projectDir: File): CompositeBazelModuleRegistryService {
+            val unknownKey = "UNKNOWN"
+
+            val packagesPerServer = urls.filter { it.endsWith("source.json") }.map {
+                val groups = URL_REGEX.matchEntire(it)?.groups
+
+                val key = groups?.get("server")?.value ?: unknownKey
+                val value = groups?.get("package")?.value ?: it
+
+                key to value
+            }.groupByTo(mutableMapOf(), { it.first }) { it.second }
+
+            packagesPerServer.remove(unknownKey)?.let {
+                logger.warn {
+                    "The following URLs cannot be mapped to a server root:"
+                }
+
+                it.forEach { logger.warn { it } }
+            }
+
+            val packagesPerRegistry = packagesPerServer.mapKeys { entry ->
+                val url = entry.key
+
+                if (url.startsWith("file://")) {
+                    val localRegistryURL = url.replace("%workspace%", projectDir.absolutePath)
+                    logger.info {
+                        "Using local Bazel module registry at '$localRegistryURL'."
+                    }
+
+                    LocalBazelModuleRegistryService(File(localRegistryURL))
+                } else {
+                    RemoteBazelModuleRegistryService.create(url)
+                }
+            }
+
+            return CompositeBazelModuleRegistryService(packagesPerRegistry)
+        }
+    }
+
+    override suspend fun getModuleMetadata(name: String): ModuleMetadata {
+        val registry = packagesPerRegistry.entries.find { name in it.value }
+            ?: throw IllegalArgumentException("No registry found for package '$name'.")
+        // TODO check if other registry entries contain the package
+
+        return registry.key.getModuleMetadata(name)
+    }
+
+    override suspend fun getModuleSourceInfo(name: String, version: String): ModuleSourceInfo {
+        val registry = packagesPerRegistry.entries.find { name in it.value }
+            ?: throw IllegalArgumentException("No registry found for package '$name'.")
+        // TODO check if other registry entries contain the package
+
+        return registry.key.getModuleSourceInfo(name, version)
```

Comment:
Hm, ff there is no result, I wonder why throw. Is this just because of the interface?

Reply:
Initially, I wanted to just output a warning like:
https://github.com/oss-review-toolkit/ort/blob/main/plugins/package-managers/bazel/src/main/kotlin/Bazel.kt#L160-L174
but a return value is needed so I had to throw an exception.

Anyway this code will change when https://github.com/oss-review-toolkit/ort/issues/8801 gets implemented.

Reply:
Would it be an option to not immplement the interface ?
Then you can just return a nullable type and the need to fail hard is gone.

Reply:
While yes it could be a reasonable solution, I fell that, when https://github.com/oss-review-toolkit/ort/issues/8801 will be implemented, all registry services will have to return null if the package is not found to allow trying more registries.
For instance currently the RemoteBazelModuleRegistryService fails with a 404 if the package is not present on the registry.

Hence, would is be ok to change the interface then ?

Reply:
> While yes it could be a reasonable solution, I fell that, when https://github.com/oss-review-toolkit/ort/issues/8801 will be implemented, all registry services will have to return null if the package is not found to allow trying more registries.

Maybe this was a misunderstanding.
1. I didn't mean to ask to change the function signature in the interface
2. Just because a return type is nullable does not mean it's forbidden to throw IMO

My reasoning was as follows: As the plan is to merge right before summer break, it would
not be nice to introduce an uncaught exception issue. (Maybe there is no and I misunderstand) 
So, to be on safe side I thought it's better to not throw for now. But it would not be written in stone. The code could be changed later on.

Reply:
Ah ok, got it. `CompositeBazelModuleRegistryService` is not even used in the caller in this case.